File: app/src/main/java/com/zawzaw/savethelibrary/utils/Moulder.java
package com.zawzaw.savethelibrary.utils;

import me.myatminsoe.mdetect.Rabbit;

/**
 * Created by zawzaw on 01/01/18.
 */

public class Moulder {

    private static boolean isUnicode;

    public static void init(boolean isUnicode) {
        Moulder.isUnicode = isUnicode;
    }

    public static String stringToUni(String text) {
        if (!isUnicode) {
            return Rabbit.zg2uni(text);
        }
        return text;
    }

    public static String mercyOnZgUser(String text) {
        if (isUnicode) {
            return text;
        } else {
            return Rabbit.uni2zg(text);
        }
    }


}

File: app/src/main/java/com/zawzaw/savethelibrary/utils/Moulder.kt
package com.zawzaw.savethelibrary.utils

import me.myatminsoe.mdetect.Rabbit

/**
 * Created by zawzaw on 01/01/18.
 */

object Moulder {

    private var isUnicode: Boolean = false

    fun init(isUnicode: Boolean) {
        Moulder.isUnicode = isUnicode
    }

    fun stringToUni(text: String): String {
        return if (!isUnicode) {
            Rabbit.zg2uni(text)
        } else text
    }

    fun mercyOnZgUser(text: String): String {
        return if (isUnicode) {
            text
        } else {
            Rabbit.uni2zg(text)
        }
    }


}
